jackson/src/main/java/feign/jackson/JacksonDecoder.java

```diff
@@ -47,7 +49,8 @@ public JacksonDecoder(ObjectMapper mapper) {
   public Object decode(Response response, Type type) throws IOException {
     if (response.body() == null)
       return null;
-    Reader reader = response.body().asReader(Util.UTF_8);
+    Charset charset = getResponseCharset(response);
+    Reader reader = response.body().asReader(charset);
```

Comment:
If you follow my additional advice below, then this could be refactored to `response.body().asReader()` where the implementation would use the charset included in the response header by default.  The result would be two methods `body().asReader()` and `body.asReader(Charset)`

Reply:
Now the method asReader() is deprecated, should i remove deprecation?